pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -158,27 +166,29 @@ func getPackagesFromCB(ctx context.Context, clusterBootstrap *runtanzuv1alpha3.C
 			if pkgShortName == "tkg-storageclass" {
 				continue
 			}
-			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: constants.TkgNamespace},
+			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: systemNamespace},
 				Spec: kapppkgv1alpha1.PackageSpec{RefName: pkgName, Version: pkgVersion}})
 		}
 	}
 
 	return packages, nil
 }
 
-// checkClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
-func checkClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
+// CheckClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
+func CheckClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
 	log.Infof("Verify addons on workload cluster %s with management cluster %s", wcClusterName, mcClusterName)
 
 	var clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap
-	if isManagementCluster {
+
+	// in tkgs infrastructure, CB only exists in the supervisor cluster
```

Comment:
The comment is not quite right. In all infrastructures (not only TKGS), CB only reside in the supervisor cluster. Only the namespace in which it exists is different for the management vs workload cluster.
For non-TKGS, the CB resource for the management cluster exists under ```tkg-system``` namespace. For the workload cluster, it exists under workload cluster's namespace in the management cluster.
For TKGS, the CB resource only exists for the guest cluster (again it resides in the supervisor cluster, but in guest cluster namespace). There is no CB for the supervisor cluster itself.

Reply:
Fixed